druid/examples/game_of_life.rs

```diff
@@ -0,0 +1,508 @@
+// Copyright 2019 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Game of life
+
+use std::ops::{Index, IndexMut};
+use std::time::{Duration, Instant};
+
+use druid::widget::{Alignment, Button, Flex, Label, Slider, WidgetExt};
+use druid::{
+    AppLauncher, BoxConstraints, Color, Data, Env, Event, EventCtx, LayoutCtx, Lens, LifeCycle,
+    LifeCycleCtx, LocalizedString, MouseButton, PaintCtx, Point, Rect, RenderContext, Size,
+    TimerToken, UnitPoint, UpdateCtx, Widget, WindowDesc,
+};
+use std::sync::Arc;
+
+const GRID_SIZE: usize = 40;
+const POOL_SIZE: usize = GRID_SIZE * GRID_SIZE;
+
+const BG: Color = Color::grey8(23 as u8);
+const C0: Color = Color::from_rgba32_u32(0xEBF1F7);
+const C1: Color = Color::from_rgba32_u32(0xA3FCF7);
+const C2: Color = Color::from_rgba32_u32(0xA2E3D8);
+const C3: Color = Color::from_rgba32_u32(0xF2E6F1);
+const C4: Color = Color::from_rgba32_u32(0xE0AFAF);
+
+#[derive(Clone, Data)]
+struct Grid {
+    storage: Arc<Vec<bool>>,
+}
+
+#[derive(Clone, Copy, PartialEq)]
+struct GridPos {
+    row: usize,
+    col: usize,
+}
+
+#[derive(Clone)]
+struct ColorScheme<'a> {
+    colors: Vec<&'a Color>,
```

Comment:
I don't think there's any need for this to be borrowed data, `Vec<Color>` should be fine. Was this about the iterator?

Reply:
yeah, i used an iterator before.

Reply:
woudn't that result in the cloning of colors every time paint is called?

Reply:
never mind, i was wrong, a simple `&self.color_scheme` solves the problem.